syncopy/connectivity/ST_compRoutines.py

```diff
@@ -399,13 +399,15 @@ class ST_CrossCovariance(ComputationalRoutine):
 
     def process_metadata(self, data, out):
 
-        # Get trialdef array + channels from source
+        # Get trialdef array + channels from source: note, since lags are encoded
+        # in time-axis, trial offsets etc. are bogus anyway: simply take max-sample
+        # counts / 2 to fit lags
         if data._selection is not None:
             chanSec = data._selection.channel
-            trl = data._selection.trialdefinition
+            trl = np.ceil(data._selection.trialdefinition / 2)
         else:
             chanSec = slice(None)
-            trl = data.trialdefinition
+            trl = np.ceil(data.trialdefinition / 2)
```

Comment:
So that was the missing piece to get it working in the end?! Was this tested with even/odd number of time-series input samples?

Reply:
Ahem, "tested" as in "I tried it even/odd sample counts". But no rigorous tests are in place yet.